### PR TITLE
docs(devs-infra): add table of contents for `README.md`

### PR DESCRIPTION
**Pull request by ahnpnl**
## Summary

Add Table of Contents for `README.md`. This will make it easier for users to search for what they want.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information
**N.A.**

### CODE REVIEW

**Comment by ahnpnl** [comment]
@wtho would you please check if it looks good ? You can view the new doc in my branch https://github.com/ahnpnl/jest-preset-angular/tree/improve-readme

File: README.md [hunk 1]
```diff
@@ -13,39 +13,88 @@ This is a part of the article: [Testing Angular faster with Jest](https://www.xf
 
 _Note: This preset does not support AngularJS (1.x). If you want to set up Jest with AngularJS, please see [this blog post](https://medium.com/aya-experience/testing-an-angularjs-app-with-jest-3029a613251)._
```
Comment (wtho) [inline README.md:14]: I would suggest to remove this part on the next major version (and the link to the article as well, as they both are pretty outdated), if thymikee is ok with it.

File: README.md [hunk 2]
```diff
@@ -86,22 +136,38 @@ module.exports = {
...
+This package provides serializer to remove such attributes. This makes snapshots cleaner and more human-readable. 
+To remove such specific attributes use `AngularNoNgAttributesSnapshotSerializer` serializer. 
+You need to add `AngularNoNgAttributesSnapshotSerializer` serializer manually (see [`test` app configuration](https://github.com/thymikee/jest-preset-angular/blob/master/e2e/test-app-v9/package.json#L47-L51)).
+
+
+## AST Transformers
+
+Jest doesn't run in browser nor through dev server. It uses jsdom to abstract browser environment. So we have to cheat 
+a little and inline our templates and get rid of styles (we're not testing CSS) because otherwise Angular will try 
+to make XHR call for our templates and fail miserably.
```
Comment (wtho) [inline README.md:165]: Note: This part is also not correct anymore after the next update. The Angular Compiler does much more than just this using ast transformers.

**Review by wtho — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (ahnpnl) [reply README.md:165]: Ye, actually once we switch to use `replaceResources` transformer, we can remove this section.

File: README.md (hunk 1 above)
Comment (thymikee) [reply README.md:14]: It's about time, let's do it 👍

Comment (ahnpnl) [reply README.md:14]: ok I will update this in #562

Comment (ahnpnl) [reply README.md:14]: added in https://github.com/thymikee/jest-preset-angular/pull/562/commits/65bedff31f5a396192c15ad916ae7d9bd2755fdd

**Pull request merged**